docs/pages/application-access/troubleshooting-apps.mdx

```diff
@@ -35,58 +35,106 @@ traffic isn't being permitted, or application logs that indicate CORS or CSRF er
 ### Solution
 
 In most cases, you can fix these types of issues by adding explicit `rewrite` settings for the Origin and Host headers
-in the Teleport configuration file for each application.
-
-To fix CSRF or CORS issues:
-
-1. Open the `/etc/teleport.yaml` file that contains the application configuration in a text editor.
-
-1. Add a `rewrite.headers` section similar to the following `grafana` example:
-
-   ```yaml
-   app_service:
-     enabled: true
-     apps:
-     - name: grafana
-       uri: http://localhost:3000
-       public_addr: grafana.teleport.example.com
-       rewrite:
-         headers:
-         - "Origin: https://grafana.teleport.example.com" # Teleport application subdomain prepended with "https://"
-         - "Host: grafana.teleport.example.com" # Teleport application subdomain itself
-   ```
-
-1. Save your changes and restart the Teleport service.
-
-To fix CSRF or CORS issues if you deploy applications using Kubernetes and `teleport-kube-agent`:
-
-1. Open the `teleport/examples/chart/teleport-kube-agent/values.yaml` file that contains the application
-   configuration in a text editor.
-
-1. Locate the `apps` section in the `values.yaml` file.
-
-   ```yaml
-   # Details of at least one app to be proxied. Example:
-   # apps:
-   #  - name: grafana
-   #    uri: http://localhost:3000
-   apps: []
-   ```
-
-1. Add a `rewrite.headers` section similar to the following `grafana` example:
-
-   ```yaml
-   app_service:
-     enabled: true
-     apps:
-     - name: grafana
-       uri: http://localhost:3000
-       public_addr: grafana.teleport.example.com
-       rewrite:
-         headers:
-         - "Origin: https://grafana.teleport.example.com" # Teleport application subdomain prepended with "https://"
-         - "Host: grafana.teleport.example.com" # Teleport application subdomain itself
-   ```
+in the Teleport configuration for each application.
+
+<Tabs>
```

Comment:
Since a user may not know which tab item is relevant until they have read all of them, I would organize the solutions into subheadings instead of tab items. How about:

```markdown
### Solution 1: Application Service configuration file
### Solution 2: `teleport-kube-agent` values file
### Solution 3: Dynamic app configuration
### Solution 4: Kubernetes app autodiscovery
```